core/src/server/Config.cpp

```diff
@@ -1013,7 +1016,12 @@ Config::GetResourceConfigIndexBuildDevice(int32_t& value) {
         return s;
     }
 
-    value = std::stoi(str.substr(3));
+    if (str == "cpu") {
+        value = CPU_DEVICE_ID;
+    } else {
+        value = std::stoi(str.substr(3));
```

Comment:
if str.substr(3) returns non-numeric string. stoi will raise an exception. Please double confirm the value of str

Reply:
"if(str == "cpu")" is used to prevent such case.